### PR TITLE
add io.github.jazztickets.irrlamb

### PR DESCRIPTION
**Pull request by jazztickets**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build io.github.jazztickets.irrlamb

**Comment by jazztickets** [comment]
bot, build io.github.jazztickets.irrlamb

File: io.github.jazztickets.irrlamb.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: https://github.com/jazztickets/irrlamb/releases/download/v1.0.2/irrlamb-1.0.2r648-src.tar.gz
+        sha256: 46c559846bbc73949682905b7c75d3b174366ed39cc57cd8ea68ff7b942a46c9
+      - type: file
+        path: io.github.jazztickets.irrlamb.appdata.xml
+    builddir: true
+
+finish-args:
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.yml:20]: finish args usually for above modules

File: io.github.jazztickets.irrlamb.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+modules:
+  - name: irrlamb
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: https://github.com/jazztickets/irrlamb/releases/download/v1.0.2/irrlamb-1.0.2r648-src.tar.gz
+        sha256: 46c559846bbc73949682905b7c75d3b174366ed39cc57cd8ea68ff7b942a46c9
+      - type: file
+        path: io.github.jazztickets.irrlamb.appdata.xml
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.yml:17]: this file is never installed.

File: io.github.jazztickets.irrlamb.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: irrlamb
+rename-desktop-file: irrlamb.desktop
+rename-icon: irrlamb
+
+modules:
+  - name: irrlamb
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.yml:11]: `cmake-ninja` doesn't work?

File: io.github.jazztickets.irrlamb.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/jazztickets/irrlamb/releases/download/v1.0.2/irrlamb-1.0.2r648-src.tar.gz
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.yml:14]: it seems that by default this tries to build with `-mfpmath=sse`... which of course doesn't work on non x86_64.

File: io.github.jazztickets.irrlamb.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	<description>
+		<p>Find all the orbs in this challenging 3D physics game, using either mouse/keyboard or gamepad.</p>
+		<p>35 levels, including 15 tutorials, 7 "campaign" levels, 4 skateparks, 7 challenges, and 2 secrets.</p>
+		<p>Keep track of your fastest times and save replays.</p>
+	</description>
+	<categories>
+		<category>Game</category>
+		<category>ArcadeGame</category>
+	</categories>
+	<content_rating/>
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.appdata.xml:15]: this is likely invalid.

File: io.github.jazztickets.irrlamb.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	<url type="homepage">https://github.com/jazztickets/irrlamb</url>
+	<url type="help">https://jazztickets.github.io/docs/irrlamb/</url>
+	<url type="bugtracker">https://github.com/jazztickets/irrlamb/issues</url>
+	<project_license>GPL-3.0</project_license>
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.appdata.xml:19]: this SPDX identifier is invalid see https://spdx.dev/licenses/

File: io.github.jazztickets.irrlamb.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        sha256: 46c559846bbc73949682905b7c75d3b174366ed39cc57cd8ea68ff7b942a46c9
+      - type: file
+        path: io.github.jazztickets.irrlamb.appdata.xml
+    builddir: true
+
+finish-args:
+  - --device=all
+  - --socket=x11
```
Comment (hfiguiere) [inline io.github.jazztickets.irrlamb.yml:22]: no wayland support I guess.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.jazztickets.irrlamb.appdata.xml (hunk 5 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.appdata.xml:15]: I ran it with `flatpak run org.freedesktop.appstream-glib validate io.github.jazztickets.irrlamb.appdata.xml` and it says OK. I'll change it to `<content_rating type="oars-1.1" />`

File: io.github.jazztickets.irrlamb.appdata.xml (hunk 6 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.appdata.xml:19]: Worked with `org.freedesktop.appstream-glib validate`, but I'll change it to `GPL-3.0-only`

File: io.github.jazztickets.irrlamb.yml (hunk 7 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.yml:22]: It's the Irrlicht engine 🤷‍♂️. Works fine in XWayland.

Comment (hfiguiere) [reply io.github.jazztickets.irrlamb.yml:22]: that's not wayland support FYI. but I'll take this as X11 only. All good.

I have no idea about the engine.

File: io.github.jazztickets.irrlamb.yml (hunk 4 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.yml:14]: SSE is for floating point determinism in the physics and replay validation system. The game doesn't need to run on ARM.

File: io.github.jazztickets.irrlamb.yml (hunk 3 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.yml:11]: I'll change it. Thanks.

File: io.github.jazztickets.irrlamb.yml (hunk 4 above)
Comment (hfiguiere) [reply io.github.jazztickets.irrlamb.yml:14]: I know what SSE is. And `-mfpmath=sse` is default on x86_64 anyway.

File: io.github.jazztickets.irrlamb.yml (hunk 1 above)
Comment (jazztickets) [reply io.github.jazztickets.irrlamb.yml:20]: Do you mean it's placed above the modules section?

**Comment by hfiguiere** [comment]
bot, build io.github.jazztickets.irrlamb

**Comment by jazztickets** [comment]
bot, build io.github.jazztickets.irrlamb

File: flathub.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+	"only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing.

Comment (jazztickets) [reply flathub.json:2]: ?

Comment (hfiguiere) [reply flathub.json:2]: that ypu chose to just ignore other arch. This is disappointing.

**Comment by jazztickets** [comment]
> Looks like it would be relatively easy to support aarch64 if you drop `-msse -mfpmath=sse`. There's probably some CMake magic to include these conditionally.

I know, but I would have to disable replay validation for that platform. That's for another time/release.

Can I just put the game on flathub for now?

**Comment by jazztickets** [comment]
I mean compiling it is one thing, but have you personally tested the game on an arm device? I doubt it'll run, being the irrlicht engine with ancient shader versions.

**Comment by hfiguiere** [comment]
can't hatch a chicken without an egg. that's kinda the situation here. There is no aarch64 build so there is no way to try.

**Pull request closed without merging**